backend/src/main/java/gov/cdc/usds/simplereport/logging/PatientExperienceLoggingInterceptor.java

```diff
@@ -0,0 +1,65 @@
+package gov.cdc.usds.simplereport.logging;
+
+import gov.cdc.usds.simplereport.api.pxp.CurrentPatientContextHolder;
+import gov.cdc.usds.simplereport.db.model.PatientLink;
+import gov.cdc.usds.simplereport.service.AuditService;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+/**
+ * HandlerInterceptor to set a request ID for patient experience REST handlers, as well as logging
+ * audit information for patient requests.
+ */
+@Component
+@ConditionalOnWebApplication
+public class PatientExperienceLoggingInterceptor implements HandlerInterceptor {
+
+  private static final Logger LOG = LoggerFactory.getLogger(PatientExperienceLoggingInterceptor.class);
+
+  @Autowired private CurrentPatientContextHolder _context;
+  @Autowired private AuditService _auditService;
+
+  @Override
+  public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
+      throws Exception {
+    LOG.debug(
+        "Pre-handling a method={} request for uri={} using handler={}",
+        request.getMethod(),
+        request.getRequestURI(),
+        handler);
+    String requestId = UUID.randomUUID().toString();
+    MDC.put(GraphQLLoggingHelpers.GRAPHQL_QUERY_MDC_KEY, requestId);
```

Comment:
Should this be added a response header here? Not sure if that would make Spring write the headers immediately, which would be not good.

Reply:
I'm fairly sure that we can add a response header here—if not, we can certainly add it in postHandle. Do we want to do the same for graphql responses?

Reply:
I thought we were already setting one on graphql responses, but it looks like I was thinking of a request header being set by the Apollo client.

Let's add the header to both response types to aid in debugging.

Reply:
Hmm. This may be a mildly vexing proposition on the graphql side, but I think it's doable. And I probably just spent somewhat longer than I should have finding out.

Reply:
Going to make adding it to graphql a separate PR.